### PR TITLE
Update Kafka and Cadence resources with PrivateLink

### PR DESCRIPTION
**Pull request by worryg0d**
Done:
- added a private link feature to the Kafka resource
- added a private link feature to the Cadence resource
- added a private link validation into Kafka and Cadence webhooks
- added datacentre number validation to the Cadence and Kafka resources

### CODE REVIEW

**Review by tengu-alt — APPROVED** [review]

**Review by testisnullus — APPROVED** [review]

File: config/samples/clusters_v1beta1_kafka.yaml [hunk 1]
```diff
@@ -56,10 +56,10 @@ spec:
 #          diskEncryptionKey: "123e4567-e89b-12d3-a456-426614174000"
 #          resourceGroup: "asdfadfsdfas"
 #      privateLink:
-#        - advertisedHostname: "asdfadsf"
+#        - advertisedHostname: "bohdan-kafka-test.com"
```
Comment (DoodgeMatvey) [inline config/samples/clusters_v1beta1_kafka.yaml:59]: I don't think that it is appropriate to use names here

File: config/samples/clusters_v1beta1_cadence.yaml [hunk 2]
```diff
@@ -51,6 +51,8 @@ spec:
       nodeSize: "CAD-DEV-t3.small-5"
       nodesNumber: 2
       clientEncryption: false
+#      privateLink:
+#        - advertisedHostname: bohdan-cadence-test.com
```
Comment (DoodgeMatvey) [inline config/samples/clusters_v1beta1_cadence.yaml:55]: I don't think that it is appropriate to use names here

File: config/samples/clusters_v1beta1_cassandra.yaml [hunk 3]
```diff
@@ -5,6 +5,7 @@ metadata:
 spec:
   name: "username-Cassandra"
   version: "4.0.10"
+  privateNetworkCluster: true
```
Comment (DoodgeMatvey) [inline config/samples/clusters_v1beta1_cassandra.yaml:8]: I think the defaul value here should be false

**Review by DoodgeMatvey — CHANGES_REQUESTED** [review]

File: config/samples/clusters_v1beta1_kafka.yaml (hunk 1 above)
Comment (worryg0d) [reply config/samples/clusters_v1beta1_kafka.yaml]: changed to `kafka-example-test.com`

File: config/samples/clusters_v1beta1_cadence.yaml (hunk 2 above)
Comment (worryg0d) [reply config/samples/clusters_v1beta1_cadence.yaml]: Fixed

File: config/samples/clusters_v1beta1_cassandra.yaml (hunk 3 above)
Comment (worryg0d) [reply config/samples/clusters_v1beta1_cassandra.yaml]: Fixed

**Review by ribaraka — APPROVED** [review]

**Review by taaraora — APPROVED** [review]

**Pull request merged**